### PR TITLE
Annotate a few NetworkX algorithm types

### PR DESCRIPTION
**Pull request by darabos**
I've never worked with Typeshed before. Apologies for any mistakes. I'm just learning this.

The types are actually nicely recorded in the docstrings for NetworkX. I'm thinking of writing a script to copy most of the types from the docstrings. At least simple int/float/str types should be doable.

### CODE REVIEW

File: stubs/networkx/networkx/algorithms/centrality/betweenness_subset.pyi [hunk 1]
```diff
@@ -1,8 +1,11 @@
-from _typeshed import Incomplete
-
+from networkx.classes.graph import Graph, _Edge, _Node
 from networkx.utils.backends import _dispatch
 
 @_dispatch
-def betweenness_centrality_subset(G, sources, targets, normalized: bool = False, weight: Incomplete | None = None): ...
+def betweenness_centrality_subset(
+    G: Graph[_Node], sources: list[_Node], targets: list[_Node], normalized: bool = False, weight: str | None = None
```
Comment (srittau) [inline stubs/networkx/networkx/algorithms/centrality/betweenness_subset.pyi:6]: One thing we do in typeshed is using more lenient annotations in arg positions, if possible:

```suggestion
    G: Graph[_Node], sources: Iterable[_Node], targets: Iterable[_Node], normalized: bool = False, weight: str | None = None
```

`Iterable` needs to be imported from `collections.abc`.

File: stubs/networkx/networkx/algorithms/centrality/betweenness_subset.pyi [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+) -> dict[_Node, float]: ...
 @_dispatch
-def edge_betweenness_centrality_subset(G, sources, targets, normalized: bool = False, weight: Incomplete | None = None): ...
+def edge_betweenness_centrality_subset(
+    G: Graph[_Node], sources: list[_Node], targets: list[_Node], normalized: bool = False, weight: str | None = None
```
Comment (srittau) [inline stubs/networkx/networkx/algorithms/centrality/betweenness_subset.pyi:10]:
```suggestion
    G: Graph[_Node], sources: Iterable[_Node], targets: Iterable[_Node], normalized: bool = False, weight: str | None = None
```

File: stubs/networkx/networkx/algorithms/centrality/closeness.pyi [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+def closeness_centrality(
+    G: Graph[_Node], u: _Node | None = None, distance: str | None = None, wf_improved: bool = True
+) -> dict[_Node, float]: ...
 @_dispatch
 def incremental_closeness_centrality(
-    G, edge, prev_cc: Incomplete | None = None, insertion: bool = True, wf_improved: bool = True
-): ...
+    G: Graph[_Node],
+    edge: _Edge[_Node],
+    prev_cc: dict[_Node, float] | None = None,
```
Comment (srittau) [inline stubs/networkx/networkx/algorithms/centrality/closeness.pyi:12]:
```suggestion
    prev_cc: SupportsKeysAndGetItem[_Node, float] | None = None,
```

`SupportsKeysAndGetItem` is a protocol defined in `_typeshed`.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks! Such a script would certainly be helpful. A few notes below.

**Review by srittau — APPROVED** [review]

**Comment by darabos** [comment]
> Thanks! Such a script would certainly be helpful. A few notes below.

Thanks for the help! I'll give it a try.

**Pull request merged**